babel/core.py

```diff
@@ -1158,13 +1201,13 @@ def parse_locale(identifier: str, sep: str = '_') -> tuple[str, str | None, str
     :raise `ValueError`: if the string does not appear to be a valid locale
                          identifier
     """
+    modifier = None
+    if '@' in identifier:
+        identifier, modifier = identifier.split('@', 1)
```

Comment:
You can use `partition` here too.
```suggestion
    identifier, _, modifier = identifier.partition('@')
```

Reply:
Sweet! I learnt something.